src/v/raft/types.h

```diff
@@ -309,6 +309,32 @@ struct snapshot_metadata {
     ss::lowres_clock::time_point cluster_time;
 };
 
+struct producer_identity {
+    int64_t id;
+    int16_t epoch;
+
+    bool operator==(const producer_identity& other) const {
```

Comment:
isn't this automatically generated?

Reply:
you have to `bool operator==(const producer_identity& other) const = default`